entrypoint.go

```diff
@@ -0,0 +1,426 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/tektoncd/pipeline/internal/artifactref"
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline"
+	v1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	"github.com/tektoncd/pipeline/pkg/internal/resultref"
+	"github.com/tektoncd/pipeline/pkg/pod"
+	"github.com/tektoncd/pipeline/pkg/result"
+	"github.com/tektoncd/pipeline/pkg/spire"
+	"github.com/tektoncd/pipeline/pkg/termination"
+	"go.uber.org/zap"
+)
+
+// RFC3339 with millisecond
+const (
+	timeFormat      = "2006-01-02T15:04:05.000Z07:00"
+	ContinueOnError = "continue"
+	FailOnError     = "stopAndFail"
+)
+
+// ScriptDir for testing
+var ScriptDir = pipeline.ScriptDir
+
+// ContextError context error type
+type ContextError string
+
+// Error implements error interface
+func (e ContextError) Error() string {
+	return string(e)
+}
+
+type SkipError string
+
+func (e SkipError) Error() string {
+	return string(e)
+}
+
+var (
+	// ErrContextDeadlineExceeded is the error returned when the context deadline is exceeded
+	ErrContextDeadlineExceeded = ContextError(context.DeadlineExceeded.Error())
+	// ErrContextCanceled is the error returned when the context is canceled
+	ErrContextCanceled = ContextError(context.Canceled.Error())
+	// ErrSkipPreviousStepFailed is the error returned when the step is skipped due to previous step error
+	ErrSkipPreviousStepFailed = SkipError("error file present, bail and skip the step")
+)
+
+// IsContextDeadlineError determine whether the error is context deadline
+func IsContextDeadlineError(err error) bool {
+	return errors.Is(err, ErrContextDeadlineExceeded)
```

Comment:
The function `IsErrSkipPreviousStepFailed` seems to be checking for `ErrContextDeadlineExceeded` instead of `ErrSkipPreviousStepFailed`. Please correct this to ensure proper error handling. [important]